### PR TITLE
Enable dev mode locally

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Tweak #3480 to only work on local dev. 

## Testing Instructions

---

Fixes #

### CODE REVIEW

File: bin/local-env/install-wordpress.sh [hunk 1]
```diff
@@ -140,3 +141,10 @@ if [ "$SCRIPT_DEBUG" != $SCRIPT_DEBUG_CURRENT ]; then
 	SCRIPT_DEBUG_RESULT=$(wp config get --type=constant --format=json SCRIPT_DEBUG | tr -d '\r')
 	echo -e $(status_message "SCRIPT_DEBUG: $SCRIPT_DEBUG_RESULT...")
 fi
+
+WEBSTORIES_DEV_MODE_CURRENT=$(wp config get --type=constant --format=json WEBSTORIES_DEV_MODE | tr -d '\r')
```
Comment (spacedmonkey) [inline bin/local-env/install-wordpress.sh:145]: I am not sure this is needed here. I think we can remove the wrapping if statement. Thoughts @swissspidy ?

**Comment by swissspidy** [comment]
Hmm not so sure about this.

This means I'd need to change `wp-config.php` on our test environment(s), and everyone not using the Docker environment would need to do so too.

I was first thinking about leaving it on `true` by default`, but when **building** the plugin setting it to `false`.

Not sure which one's better.

Thoughts?

File: bin/local-env/install-wordpress.sh (hunk 1 above)
Comment (swissspidy) [reply bin/local-env/install-wordpress.sh:145]: This keeps it in line with similar constructs above for `WP_DEBUG` and `SCRIPT_DEBUG`, so I think it's fine.

Also, helps avoid making unneeded calls to the Docker container.

**Comment by spacedmonkey** [comment]
I personally believe that `WEBSTORIES_DEV_MODE` is a feature flag itself. We should not opt people into dev mode. Tweaking dev env is a small ask, IMO. Jetpack for example, `JETPACK_DEV_DEBUG ` checks if it defined and do not opt in. 

We can document `WEBSTORIES_DEV_MODE` similar to [this](https://github.com/Automattic/jetpack/blob/4392a22d643eaf812f42c37739bf675ea3ef7f34/docs/development-environment.md#jetpack_dev_debug).

**Comment by swissspidy** [comment]
> We should not opt people into dev mode

Agreed, that's why I would have toggled the flag for all production builds we send to users.

But if you `git clone` the repository and install the plugin, you'd automatically have it enabled.

**Comment by spacedmonkey** [comment]
>But if you git clone the repository and install the plugin, you'd automatically have it enabled. 

I don't have a problem with that. If you are git cloning a repo into say VVV, you should really know what you are doing. But how about this. 

```
if ( ! defined( 'WEBSTORIES_DEV_MODE' ) ) {
	define( 'WEBSTORIES_DEV_MODE', WP_DEBUG );
}
```
If you have debug on, then you normally in a local env and are a developer. Thoughts?

**Comment by swissspidy** [comment]
Then I'd prefer `false` as the default.

Now I just need to fix the tests caused by this change.

**Pull request merged**